snapcraft/project/_environment_checks.py

```diff
@@ -0,0 +1,132 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import List, Optional  # noqa: F401
+from typing import Dict
+
+from snapcraft import project
+from snapcraft.internal import common, errors, os_release
+
+_MSG_NO_BASE_FOR_SNAP = (
```

Comment:
If using a snap (new enough to get this check), the no-base check will be invoked from the 3.x version before calling into this legacy mode.  Would it make more sense to just provide this error there?

Reply:
The checks on non legacy will be much more lighter, we don't print any message there so things should be mostly fine (and when bases become mandatory, we can throw away the old stuff more easily)